Foundation/JSONSerialization.swift

```diff
@@ -336,92 +366,33 @@ private struct JSONWriter {
         }
     }
 
-    private func serializeUInt(value: UInt) throws {
-        if value == 0 {
-            writer("0")
-            return
-        }
-        var array: [UInt] = []
-        var stringResult = ""
-        //Maximum length of an UInt
-        array.reserveCapacity(maxUIntLength)
-        stringResult.reserveCapacity(maxUIntLength)
-        var number = value
-        
-        while number != 0 {
-            array.append(number % 10)
+    private func serializeInteger<T: UnsignedInteger>(value: T, isNegative: Bool = false) {
+        let maxIntLength = 22   // 20 digits in UInt64 + optional sign + trailing '\0'
+        let ZERO: CChar = 0x30  // ASCII '0' == 0x30
```

Comment:
From a purely stylistic standpoint, I don't think we really use all caps anywhere to indicate a constant. I think this would read more clearly as just `asciiZero` (and `MINUS` would read a bit better as `asciiMinus`).

Reply:
I actually copied this from `Decimal` but I will tidy it up.